packages/cli/src/License.ts

```diff
@@ -26,18 +28,28 @@ export class License {
 
 	private manager: LicenseManager | undefined;
 
+	instanceId: string | undefined;
+
+	private redisPublisher: RedisServicePubSubPublisher;
+
 	constructor() {
 		this.logger = getLogger();
 	}
 
-	async init(instanceId: string) {
-		if (this.manager) {
+	async init(instanceId: string, instanceType: N8nInstanceType = 'main') {
+		const isMainInstance = instanceType === 'main';
+		if (this.manager && isMainInstance) {
```

Comment:
why is `&& isMainInstance` needed here?

Reply:
because we dont want to redo the init() on the main instances (since they were the onces who actually did the license renewal), only on the workers. so if the manager was already initialised on main, just return

if we can use the reload() instead of init(), maybe we don't even need all this